tests/bsim/bluetooth/mesh/src/friendship_common.c

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "friendship_common.h"
+#include "mesh_test.h"
+
+#define LOG_MODULE_NAME friendship_common
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(LOG_MODULE_NAME);
+
+uint16_t friend_lpn_addr;
+ATOMIC_DEFINE(state, TEST_FLAGS);
+struct k_sem events[TEST_FLAGS];
+
+void evt_signal(enum test_flags evt)
+{
+	atomic_set_bit(state, evt);
+	k_sem_give(&events[evt]);
+}
+
+int evt_wait(enum test_flags evt, k_timeout_t timeout)
+{
+	return k_sem_take(&events[evt], timeout);
+}
+
+void evt_clear(enum test_flags evt)
+{
+	atomic_clear_bit(state, evt);
+	k_sem_reset(&events[evt]);
+}
```

Comment:
I agree, I should've give it more thought. I think good name for them would be something like `friendship_evt_wait`, because `friend` might indicate that it's waiter for Friend events only, where it can wait for LPN event's also. Same of course goes for others, as you mentioned.